### PR TITLE
osd/osdmap Incorrect map member initialization

### PR DESCRIPTION
**Pull request by cxytt**
see tracker: http://tracker.ceph.com/issues/37501#change-125671

Signed-off-by: ningtao <ningtao@sangfor.com.cn>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/osd/OSDMap.cc [hunk 1]
```diff
@@ -4419,7 +4419,9 @@ int OSDMap::calc_pg_upmaps(
 	      ldout(cct, 10) << "  dropping pg_upmap_items " << pg
 			     << " " << p->second << dendl;
               for (auto i : p->second) {
-                pgs_by_osd[i.second].erase(pg);
+                if (pgs_by_osd.count(i.second)) {
+                  pgs_by_osd[i.second].erase(pg);
```
Comment (batrick) [inline src/osd/OSDMap.cc]:
````
if (auto it = pgs_by_osd.find(i.second); it != pgs_by_osd.end()) {
  pgs_by_osd.erase(it);
 }
````

Would be more efficient.

**Comment by cxytt** [comment]
@xiexingguo 
I'm sorry.At present, my environment cannot compile master, I wonder if you can help verify it.The steps are simple.

File: src/osd/OSDMap.cc (hunk 1 above)
Comment (cxytt) [reply src/osd/OSDMap.cc]: All right. I will modify it.

**Comment by xiexingguo** [comment]
@cxytt See https://github.com/ceph/ceph/pull/25630

**Comment by xiexingguo** [comment]
https://github.com/ceph/ceph/pull/25630 merged

**Pull request closed without merging**